lib/Dialect/TTNN/Analysis/CMakeLists.txt

```diff
@@ -15,6 +15,6 @@ add_mlir_dialect_library(MLIRTTNNAnalysis
         MLIRTTNNPassesIncGen
         MLIRTTOpsIncGen
 
-        LINK_LIBS
+        LINK_LIBS PUBLIC
```

Comment:
Just to give me context, why is this needed?

Reply:
My understanding is that is similar to c++ public inheritance. Let's say we want to link lib A into lib B and then lib B into C. If we assume that the type of linking is PUBLIC for both libs, the lib C should implicitly link the lib C. @mtopalovicTT